CommonTools/PileupAlgos/plugins/PuppiPhoton.cc

```diff
@@ -76,7 +71,25 @@ void PuppiPhoton::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
   edm::Handle<CandidateView> hPuppiProduct;
   iEvent.getByToken(tokenPuppiCandidates_,hPuppiProduct);
   const CandidateView *pupCol = hPuppiProduct.product();
-  for(CandidateView::const_iterator itPho = phoCol->begin(); itPho!=phoCol->end(); itPho++) {
+  if(usePFphotons_) {
+   for(CandidateView::const_iterator itPho = pfCol->begin(); itPho!=pfCol->end(); itPho++) {
```

Comment:
the line "phoCands.push_back(&(*pho));" requires a pointer as input. I don't see how I could obtain this from the reference pho. "phoCands.push_back(&pho);" would compile but not build (since it may give a pointer to a View?). Not sure how to do this.